src/d/a/obj/d_a_obj_tbox.cpp

```diff
@@ -0,0 +1,2673 @@
+bool dAcTbox_c::checkTboxFlag() const {
+    if (field_0x1207 <= 0x1F) {
+        return TBoxflagManager::sInstance->checkFlag(TBoxflagManager::sInstance->mSceneIndex, field_0x1207);
+    }
+    return false;
+}
+
+bool dAcTbox_c::isNotSmall() const {
+    return mVariant != SMALL;
+}
+
+bool dAcTbox_c::fn_8026D540() {
+    return field_0x11F0 == 1;
+}
+
+bool dAcTbox_c::fn_8026D560() const {
+    if (isValidGroupIndex(field_0x1208)) {
+        u32 counterVal;
+        getSomeCounter(&counterVal);
+        f32 cmp = 1.0f / counterVal;
+        return cM::rndF(1.0f) <= cmp;
+    }
+    return true;
+}
+
+bool dAcTbox_c::getSomeCounter(u32 *outValue) const {
+    bool ok = isValidGroupIndex(field_0x1208);
+    if (ok) {
+        *outValue = sSomeCounters[field_0x1208];
+        ok = *outValue;
+    }
+    return ok;
+}
+
+bool dAcTbox_c::fn_8026D670() const {
+    return isValidGroupIndex(field_0x1208);
+}
+
+void dAcTbox_c::syncScaleToMdl(m3d::scnLeaf_c *lf) {
+    lf->setScale(mScale);
+}
+
+bool dAcTbox_c::initBgW(dBgW &bg, const char *arcName, const char *fileName) {
+    void *dzb = getOarcDZB(arcName, fileName);
+    if (dzb == nullptr) {
+        return false;
+    }
+    void *plc = getOarcPLC(arcName, fileName);
+    if (plc == nullptr) {
+        return false;
+    }
+    // TODO return types
+    bool ok = !(BOOL)bg.Set((cBgD_t *)dzb, (PLC *)plc, 0x81, &mWorldMtx, &mScale);
+    if (!ok) {
+        return false;
+    }
+    bg.SetCrrFunc(dBgS_MoveBGProc_TypicalRotY);
+    return true;
+}
+
+void dAcTbox_c::initDowsingTarget(DowsingTarget *target, DowsingTarget::DowsingSlot slot, mVec3_c *offset) {
+    target->initialize(slot, 0, offset, 0.0f);
+}
+
+bool dAcTbox_c::isItemRupee() const {
+    return dAcItem_c::isRupee((ITEM_ID)mItemId);
+}
+
+void dAcTbox_c::initDowsingTarget(DowsingTarget::DowsingSlot slot) {
+    mVec3_c offset;
+    getDowsingTargetOffset(&offset);
+    initDowsingTarget(&mDowsingTarget, slot, &offset);
+}
+
+void dAcTbox_c::initDowsingTargetCube() {
+    mVec3_c offset;
+    getDowsingTargetOffset(&offset);
+    initDowsingTarget(&mGoddessDowsingTarget, DowsingTarget::SLOT_GODDESS_CUBE, &offset);
+}
+
+void dAcTbox_c::noRegisterDowsing() {}
+
+void dAcTbox_c::registerKeyPieceDowsing() {
+    if ((field_0x11F4 & 4) == 0 && StoryflagManager::sInstance->getCounterOrFlag(0x6A)) {
+        mDowsingTarget.doRegister();
+        field_0x11F4 |= 4;
+    }
+}
+
+void dAcTbox_c::registerRupeeOrTreasureDowsing() {
+    if ((field_0x11F4 & 4) == 0) {
+        mDowsingTarget.doRegister();
+        field_0x11F4 |= 4;
+    }
+}
+
+void dAcTbox_c::noUnregisterDowsing() {}
+
+void dAcTbox_c::unregisterDowsing() {
+    if ((field_0x11F4 & 4) != 0) {
+        mDowsingTarget.doUnregister();
+        field_0x11F4 &= ~4;
+    }
+}
+
+extern "C" u16 PARTICLE_RESOURCE_ID_MAPPING_208_;
+
+void dAcTbox_c::fn_8026D950() {
+    fn_800298B0(PARTICLE_RESOURCE_ID_MAPPING_208_, &position, &rotation, nullptr, nullptr, nullptr, nullptr, nullptr);
+}
```

Comment:
some appear particle effect?
![image](https://github.com/user-attachments/assets/db59b027-ddff-4484-9916-3c189113093d)

Reply:
if you are every curious, uploading common.jpc to noclip.website can display a large amount of the particles.